src/utils/api.js

```diff
@@ -0,0 +1,18 @@
+import axios from "axios";
+
+const newsApi = axios.create({baseURL : "https://first-project-svt2.onrender.com/api"})
+export const getArticles=(query)=>{
+    let article = '/articles'
+    if(query!== null){
+        article+=`?topic=${query}`
```

Comment:
You don't need to do so much work here to check if the query is there or not 🧠
https://axios-http.com/docs/example
The docs explain that we can do /api/user?id=234 using a options object as the second argument. Have a look through how they do it but this screenshot is also pretty important
<img width="636" alt="image" src="https://github.com/SuganyaArul/nc_news_front_end/assets/151854815/00be6fc8-0029-470a-90ca-9708873a28b2">

Here we're saying that if you use the params in a options object and they're undefined then axios won't put them in the URL, creating an optional query